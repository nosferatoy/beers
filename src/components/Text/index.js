import React from 'react';
import cx from 'classnames';

import './index.scss';

const Text = ({ size, color, opacity, bold, content }) => {
	const newClassName =
		(size ? ` text-${size}` : '') +
		(color ? ` text-${color}` : '') +
		(opacity ? ` opacity-${opacity}` : '') +
		(bold ? ` bold-${bold}` : '');

	return <div className={cx(newClassName)}>{content}</div>;
};
export const Title = props => <Text color="light" size="xxlarge" {...props} />;
Text.Title = Title;

export default Text;
