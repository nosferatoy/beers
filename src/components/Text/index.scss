$baseFontSize: 16;

$textSizes: (
  xxsmall:10,
  xsmall: 12,
  small: 14,
  normal: 16,
  slarge: 18,
  large: 21,
  xlarge: 24,
  xxlarge: 26,
  title: 30,
  xtitle: 36,
  sheading: 45,
  heading: 50,
  xheading: 60
);

$colors: (
  light: var(--color-ligth),
  primary: var(--color-primary),
  secondary: var(--color-secondary),
  light-gray: var(--color-light-gray),
  dark: var(--color-dark),
  danger: var(--color-danger-dark),
  light-green: var(--color-light-green)
);

$bold : (
  "black": 900,
  "heavy": 800,
  "bold": 700,
  "semibold": 600,
  "medium": 500,
  "regular": 400,
  "thin": 300,
  "light": 200,
  "ultralight": 100
);

$opacity : (
  "transparent": 0,
  "xlow": 0.1,
  "low": 0.3,
  "medium": 0.5,
  "high": 0.7,
  "xhigh": 0.9,
  "opaque": 1
);

@each $name, $value in $textSizes {
  .text-#{$name} {
    font-size: #{$value/$baseFontSize}rem;
  }
}

@each $name, $value in $colors {
  .text-#{$name} {
    color: $value;
  }
}

@each $name, $value in $opacity {
  .opacity-#{$name} {
    opacity: $value ;
  }
}

@each $name, $value in $bold {
  .bold-#{$name} {
    font-weight: $value ;
  }
}


