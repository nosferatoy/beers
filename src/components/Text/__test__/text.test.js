import React from 'react';
import { render } from '@testing-library/react';
import Text, { Title } from '../index';

describe('Test Text component', () => {
	it('Render Text', () => {
		const { container } = render(<Text content="Normal text" />);
		expect(container).toMatchSnapshot();
	});

	it('Render Text - bold and color', () => {
		const { container } = render(
			<Text bold="bold" color="color-primary" content="Bold color text" />
		);
		expect(container).toMatchSnapshot();
	});

	it('Render Text Title', () => {
		const { container } = render(<Title content="Title text" />);
		expect(container).toMatchSnapshot();
	});
});
