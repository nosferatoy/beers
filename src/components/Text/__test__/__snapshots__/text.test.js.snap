// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Text component Render Text - bold and color 1`] = `
<div>
  <div
    class=" text-color-primary bold-bold"
  >
    Bold color text
  </div>
</div>
`;

exports[`Test Text component Render Text 1`] = `
<div>
  <div
    class=""
  >
    Normal text
  </div>
</div>
`;

exports[`Test Text component Render Text Title 1`] = `
<div>
  <div
    class=" text-xxlarge text-light"
  >
    Title text
  </div>
</div>
`;
